analyses/molecular-subtyping-CRANIO/00-craniopharyngiomas-molecular-subtype.Rmd

```diff
@@ -0,0 +1,180 @@
+
+CRANIO, PAP
+- Contains BRAF V600E mutation
+- Tumor occurs exclusively in adults (40+ years)
+
+# Usage
+
+This notebook is intended to be run via the command line from the top directory
+of the repository as follows:
+
+```
+Rscript -e "rmarkdown::render('analyses/molecular-subtyping-CRANIO/00-craniopharyngiomas-molecular-subtype.Rmd', clean = TRUE)"
+```
+
+# Set up
+
+```{r load-library}
+library(tidyverse)
+```
+
+## Directories and File Inputs/Outputs
+
+```{r set-dir-and-file-names}
+# Detect the ".git" folder -- this will in the project root directory.
+# Use this as the root directory to ensure proper sourcing of functions no
+# matter where this is called from
+root_dir <- rprojroot::find_root(rprojroot::has_dir(".git"))
+# File path to results directory
+results_dir <-
+  file.path(root_dir, "analyses", "molecular-subtyping-CRANIO", "results")
+if (!dir.exists(results_dir)) {
+  dir.create(results_dir)
+}
+
+# Outputs
+output_lesions <- file.path(results_dir, "CRANIO_defining_lesions.tsv")
+output_subtype <- file.path(results_dir, "CRANIO_molecular_subtype.tsv")
+```
+
+```{r load-inputs-please-wait}
+# Inputs
+## Read in metadata
+metadata <-
+  read.table(file.path(root_dir, "data", "pbta-histologies.tsv"),
+             header=T,
+             sep='\t') %>%
+  filter(pathology_diagnosis == "Craniopharyngioma")
+## Read in snv consensus mutation data
+snv_df <-
+  read.table(file.path(root_dir, "data", "pbta-snv-consensus-mutation.maf.tsv.gz"),
+             header=T,
+             sep='\t')
+```
+
+# Process Data
+
+## Collecting Tumor Sample Barcodes with Variants of Interest
+
+Collect Tumor_Sample_Barcodes with BRAF V600E mutations.
+
+```{r braf-tsb}
+braf_v600e_tsb <-
+  snv_df %>%
+  filter(Hugo_Symbol == "BRAF" & HGVSp_Short == "p.V600E") %>%
+  select(Tumor_Sample_Barcode)
+```
+
+Collect Tumor_Sample_Barcodes with CTNNB1 mutation on exon 3.
+
+```{r ctnnb1-tsb}
+ctnnb1_e3_tsb <-
+  snv_df %>%
+  filter(Hugo_Symbol == "CTNNB1" & Exon_Number == "3/15") %>%
+  select(Tumor_Sample_Barcode)
+```
+
+## Defining Lesion Table
+
+Annotate the metadata table where the Kids_First_Biospecimen_ID matches either of the TSBs collected above
+Lesions are defined where:
+1. patient has BRAF.V600E mutation and is older than the age_of_adulthood (40 years in this case)
+1. patient has CTNNB1.Exon3 mutation and is younger than the age_of_adulthood (40 years in this case)
+Select the columns from the table that the PIs have requested
+
+```{r snv-lesion-table}
+# Dispute this all you like
+age_of_adulthood <- 40 * 365.25
+
+snv_lesion_df <-
+  metadata %>%
+  mutate(CTNNB1.Exon3 = Kids_First_Biospecimen_ID %in% ctnnb1_e3_tsb$Tumor_Sample_Barcode) %>%
+  mutate(BRAF.V600E = Kids_First_Biospecimen_ID %in% braf_v600e_tsb$Tumor_Sample_Barcode) %>%
+  mutate(defining_lesion =
+           (BRAF.V600E & as.numeric(as.character(age_at_diagnosis_days)) >= age_of_adulthood) |
```

Comment:
This step is a lot to digest and interpret as far as what `defining_lesion` represents. Do you think you could split this up into smaller steps? In other words, `as.numeric(as.character(age_at_diagnosis_days))` could be its own step as well as the comparison to `age_of_adulthood` could be its own variable.

Reply:
Now that I see you are doing the main `case_when` later, I think this step could really just be simplified to determining age only and then your `case_when` can determine the final call? Let me know if I'm misunderstanding what `defining_lesion` represents though.

Reply:
Ok. I removed the bit on defining_lesion from this table here and re-implemented it later in the case when statement. This mutation was a bool check of is there either a BRAF mutation in adults or a CTNNB1 mutation in children. This simplification makes sense now as it can only be one or the other (people can't be simultaneously a child and adult at diagnosis) but if the age requirements for these diverge it might be possible.